sos/plugins/__init__.py

```diff
@@ -819,6 +830,21 @@ def getmtime(path):
                 except OSError:
                     return 0
 
+            def time_filter(path):
+                """ When --since is passed, or maxage is coming from the
+                plugin, we need to filter out older files """
+
+                if archive_pattern.search(path) is None:
```

Comment:
What is the purpose of `archive_pattern`? Current implementation means that `--since` is applied *only* on files matching the pattern. So if I edit auditd plugin to collect `/var/log/audit/*`, then `sosreport --since 20190808 --all-logs --batch --build` will collect also `/var/log/audit/audit.log.4` with mtime 4 days in past.

I agree we shall apply the `--since` to (logrotated) log files only - not to current log file (well this could be arguable) and definitely `--since` cant be applied to config files.

Or maybe `archive_pattern` just needs adding something like `log.[1-9]`? E.g.:

    archive_pattern = re.compile('.*(.zip|.gz|.bz2|.xz|[-.][0-9]+)$')

Or even better:

    archive_pattern = re.compile('.*.log.+$')

to cover *any* logrotated log file? (this can be too generic and we can skip collecting some config file like `/etc/foo/bar.logging.conf` - I prefer the prior variant)

Reply:
Improved suggestion:

    archive_pattern = re.compile('.*(\.zip|\.gz|\.bz2|\.xz|[\.-]\d+)$')

(`.` to `\.` and `[0-9]` to `\d`)

Reply:
The purpose of the `archive_pattern` is that, even if we `--since`, we want to include more static log files (`dmesg` for example). We also don't want to affect `/etc`.

If we `'.*.log.+$'`, some logs like `(httpd/(access|error)_log|messages)-XXXXXXXX` won't be filtered. 

I think the most bulletproof approach would be parsing the `logrotate`'s config, but that would add some overhead.

I like the idea of `[-.][0-9]+)$`, that should fix it for `audit` if we change the `copyspecs` passed to `add_copy_spec`.